src/main/telemetry/smartport.c

```diff
@@ -288,7 +288,11 @@ void smartPortSendByte(uint8_t c, uint16_t *checksum, serialPort_t *port)
 
 bool smartPortPayloadContainsMSP(const smartPortPayload_t *payload)
 {
-    return payload->frameId == FSSP_MSPC_FRAME_SMARTPORT || payload->frameId == FSSP_MSPC_FRAME_FPORT;
```

Comment:
maybe: 
``` 
return payload && (payload->frameId == FSSP_MSPC_FRAME_SMARTPORT || payload->frameId == FSSP_MSPC_FRAME_FPORT);
```

It may be preferable to move the `NULL` test out, passing only valid payload to this function

Reply:
In both places where this function is used, a NULL check is required, so I think it makes sense to do it in the function.

Reply:
It's mostly about style - 
With the  test around the call, it is obvious that `payload` may be empty. This may prevent bug fixed by this PR in future.
Even keeping test inside as safeguard and duplicating it before call is IMO reasonable.

From FLASH point of view, it is only one or two instructions difference if call is made and compiler does probably inline the function, removing all penalty.

Reply:
@ledvinap that short version definitely look better

Reply:
@ledvinap ok ill place NULL test outside, the name of function is smartPortPayloadContainsMSP, not smartPortPayloadIsValidAndContainsMSP

Reply:
@ledvinap: Good point.